pkg/api/controller.go

```diff
@@ -170,11 +175,46 @@ func (c *Controller) Logout(w http.ResponseWriter, _ *http.Request) {
 		Expires:  time.Unix(0, 0),
 		SameSite: http.SameSiteStrictMode,
 	})
-
+	err := doOIDCLogout(w, r, c.sessionStore)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
 	writeResponse(w, http.StatusOK, nil)
 }
 
+// OauthCallback gets a code generated by an OIDC provider.
+// It exchanges the code for an id token, and saves the claims from the ID token on a session.
+func (c *Controller) OauthCallback(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	session, err := c.sessionStore.Get(r, OIDCAuthSessionName)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	if r.URL.Query().Get("state") != session.Values[StateSessionKey] {
+		writeError(w, http.StatusBadRequest, "Invalid state parameter.")
+		return
+	}
+	idTokenClaims, err := c.oidcAuthenticator.GetIDTokenClaims(ctx, r.URL.Query().Get("code"))
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	session.Values[IDTokenClaimsSessionKey] = idTokenClaims
+	err = session.Save(r, w)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	http.Redirect(w, r, "/", http.StatusTemporaryRedirect)
+}
+
 func (c *Controller) Login(w http.ResponseWriter, r *http.Request, body LoginJSONRequestBody) {
+	err := doOIDCLogout(w, r, c.sessionStore)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to perform OIDC logout")
+	}
```

Comment:
Think we need to change one of two places. We can ignore the error here as loging a session in a system that doesn't have a session will fail. Or ignore it as we already just log it.

Reply:
Fixed: now ignoring the error from `sessionStore.Get` when it makes sense.